src/v/cluster/id_allocator_frontend.cc

```diff
@@ -45,50 +45,88 @@ id_allocator_frontend::id_allocator_frontend(
   , _metadata_cache(metadata_cache)
   , _connection_cache(connection_cache)
   , _leaders(leaders)
-  , _controller(controller) {}
+  , _controller(controller)
+  , _metadata_dissemination_retries(
+      config::shard_local_cfg().metadata_dissemination_retries.value())
+  , _metadata_dissemination_retry_delay_ms(
+      config::shard_local_cfg().metadata_dissemination_retry_delay_ms.value()) {
+}
+
+ss::future<std::optional<model::node_id>> id_allocator_frontend::get_leader() {
+    auto leader = _leaders.local().get_leader(model::id_allocator_ntp);
+
+    if (unlikely(!leader)) {
+        auto retries = _metadata_dissemination_retries;
+        auto delay_ms = _metadata_dissemination_retry_delay_ms;
+        while (!leader && 0 < retries--) {
+            co_await ss::sleep(delay_ms);
```

Comment:
use `sleep_abortable()` since this is user passed, it can be a long time and if we hit Ctrl-C the application may 'hang'